CoEpi/ble/Peripheral.swift

```diff
@@ -1,33 +1,43 @@
-import Foundation
 import CoreBluetooth
+import Dip
+import Foundation
 import os.log
 import RxRelay
 
+struct Contact2 {
+    let theirData: Data?
+    let ourData: Data?
+}
 
 protocol PeripheralDelegate: class {
     func onPeripheralStateChange(description: String)
-    func onPeripheralContact(_ contact: CEN)
+    func onPeripheralContact(_ contact: Contact2)
 }
 
 protocol PeripheralReactive {
     var peripheralState: BehaviorRelay<String> { get }
 
-    var peripheralContactSent: PublishRelay<ContactOld> { get }
+    var peripheralContactSent: PublishRelay<Contact2> { get }
     var delegate: PeripheralDelegate? {get set}
 }
 
+protocol PeripheralRequestHandler: class {
+    func onReceivedRequest(request: Data?, respondClosure: (Data) -> ())
+}
 
 class PeripheralImpl: NSObject, PeripheralReactive {
     let peripheralState: BehaviorRelay<String> = BehaviorRelay(value: "unknown")
-    let peripheralContactSent: PublishRelay<ContactOld> = PublishRelay()
+    let peripheralContactSent: PublishRelay<Contact2> = PublishRelay()
     weak var delegate: PeripheralDelegate?
-
+    private let internalDelegate: PeripheralRequestHandler
```

Comment:
What is the point of adding another delegate? There is already var delegate: PeripheralDelegate

Reply:
That delegate handles when the peripheral is connected to, aka when we generate keys and save to DB. The point is so that no CEN/DB code lives in the peripheral - the peripheral just calls some delegate.